File: DevExpress VCL/ExpressQuantumTreeList 4/Demos/Delphi/SummariesDemo/SummariesDemoData.pas
unit SummariesDemoData;

interface

uses
  SysUtils, Classes, DB, Forms, cxClasses, cxStyles, cxTL, ImgList,
  Controls, DBTables;

type
  TSummariesDemoDataDM = class(TDataModule)
    DataBase: TDataBase;
    tblOrders: TTable;
    dsOrders: TDataSource;
    tblOrdersID: TAutoIncField;
    tblOrdersCustomerID: TIntegerField;
    tblOrdersProductID: TIntegerField;
    tblOrdersPurchaseDate: TDateTimeField;
    tblOrdersTime: TDateTimeField;
    tblOrdersPaymentType: TStringField;
    tblOrdersPaymentAmount: TFloatField;
    tblOrdersQuantity: TIntegerField;
    tblCars: TTable;
    dsCars: TDataSource;
    tblCustomers: TTable;
    dsCustomers: TDataSource;
    tblCustomersID: TAutoIncField;
    tblCustomersFirstName: TStringField;
    tblCustomersLastName: TStringField;
    tblCustomersName: TStringField;
    tblCarsID: TAutoIncField;
    tblCarsTrademark: TStringField;
    tblCarsModel: TStringField;
    tblCarsPicture: TBlobField;
    tblCarsCarName: TStringField;
    StyleRepository: TcxStyleRepository;
    cxStyle1: TcxStyle;
    cxStyle2: TcxStyle;
    cxStyle3: TcxStyle;
    cxStyle4: TcxStyle;
    cxStyle5: TcxStyle;
    cxStyle6: TcxStyle;
    cxStyle7: TcxStyle;
    cxStyle8: TcxStyle;
    cxStyle9: TcxStyle;
    cxStyle10: TcxStyle;
    cxStyle11: TcxStyle;
    cxStyle12: TcxStyle;
    cxStyle13: TcxStyle;
    stlGroupNode: TcxStyle;
    stlFixedBand: TcxStyle;
    TreeListStyleSheetDevExpress: TcxTreeListStyleSheet;
    tblCarsDescription: TMemoField;
    tblCarsPrice: TFloatField;
    PaymentTypeImages: TImageList;
    qryOrdersSummary: TQuery;
    dsOrdersSummary: TDataSource;
    procedure tblCustomersCalcFields(DataSet: TDataSet);
    procedure tblCarsCalcFields(DataSet: TDataSet);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  SummariesDemoDataDM: TSummariesDemoDataDM;

implementation

{$R *.dfm}

procedure TSummariesDemoDataDM.tblCustomersCalcFields(DataSet: TDataSet);
begin
  tblCustomersName.Value := tblCustomersFirstName.AsString + ' ' +
    tblCustomersLastName.AsString;
end;

procedure TSummariesDemoDataDM.tblCarsCalcFields(DataSet: TDataSet);
begin
  tblCarsCarName.Value := tblCarsTrademark.AsString + ' ' +
    tblCarsModel.AsString;
end;

end.

File: DevExpress VCL/ExpressQuantumTreeList 4/Demos/CBuilder/SummariesDemo/SummariesDemoData.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "SummariesDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxStyles"
#pragma link "cxTL"
#pragma resource "*.dfm"
TSummariesDemoDataDM *SummariesDemoDataDM;
//---------------------------------------------------------------------------
__fastcall TSummariesDemoDataDM::TSummariesDemoDataDM(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TSummariesDemoDataDM::tblCustomersCalcFields(TDataSet *DataSet)
{
  tblCustomersName->Value = tblCustomersFirstName->AsString + " " +
    tblCustomersLastName->AsString;
}
//---------------------------------------------------------------------------

void __fastcall TSummariesDemoDataDM::tblCarsCalcFields(TDataSet *DataSet)
{
  tblCarsCarName->Value = tblCarsTrademark->AsString + " " +
    tblCarsModel->AsString;
}
//---------------------------------------------------------------------------


